### PR TITLE
Added separation among UpdateVersion and UpdateConfig in NodePool Inplace Upgrade

### PR DESCRIPTION
**Pull request by jparrill**
Implemented a separation in status.conditions to identify when a ConfigChange vs OpenshiftUpgrade happens in the InPlace upgrade type.

The behaviour it's the next:

Regarding UpdateConfig:
- If there is a change in progress, a new status.condition appears: UpdatingConfig
- It will show UPDATINGCONFIG as True in the oc get interface
- This status will comeback to False when the ConfigChange finishes

Regarding UpdateVersion:
- If you modify the NodePool release, a new field in NodePool.Status.Conditions appears: UpdatingVersion
- It will show UPDATINGCONFIG and UPDATINGVERSION in the oc get interface
- These statuses will comeback to False when the Upgrade finishes

NOTE: Kudos to @enxebre for helping me in the implementation.

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

**Which issue(s) this PR fixes**:

Fixes #[HOSTEDCP-403](https://issues.redhat.com/browse/HOSTEDCP-403)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by jparrill** [comment]
Some screenshots in local tests:

Updating Config:
Screenshot 2022-11-30 at 12 50 13
Screenshot 2022-11-30 at 12 53 05

Updating Release:
Screenshot 2022-11-30 at 12 55 46
Screenshot 2022-11-30 at 12 55 31

Once both ends:
Screenshot 2022-11-30 at 13 01 50

File: hypershift-operator/controllers/nodepool/inplace.go [hunk 1]
```diff
@@ -151,32 +152,37 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
...
+		})
+		log.Info("NodePool config is updating",
+			"current", nodePool.GetAnnotations()[nodePoolAnnotationCurrentConfig],
+			"target", targetConfigHash)
+	} else {
+		removeStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolUpdatingConfigConditionType)
 	}
 
-	if _, ok := machineSet.Annotations[nodePoolAnnotationUpgradeInProgressFalse]; ok {
-		status = corev1.ConditionFalse
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/inplace.go:168]: This logic is necessary to report reason and message, why are we removing it?

Comment (enxebre) [reply hypershift-operator/controllers/nodepool/inplace.go:168]: See https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go#L45-L51

Comment (jparrill) [reply hypershift-operator/controllers/nodepool/inplace.go:168]: This applies also for configUpdate? (not just the releaseUpdate), I think so, but just to confirm it.

File: hypershift-operator/controllers/nodepool/inplace.go [hunk 2]
```diff
@@ -169,7 +168,14 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
...
 		message = machineSet.Annotations[nodePoolAnnotationUpgradeInProgressFalse]
 	}
-	if message != "" {
+
+	// Check if config needs to be updated.
+	isUpdatingConfig := isUpdatingConfig(nodePool, targetConfigHash)
+
+	// Check if version needs to be updated.
+	isUpdatingVersion := isUpdatingVersion(nodePool, targetVersion)
+
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/inplace.go:177]: thanks, this is a small improvement. Can we please follow up in another PR to:
Make sure the conditions are always set either false/true/unknown, regardless of message being empty or not.

**Pull request merged**